sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -151,7 +151,7 @@ statement
     | ALTER TABLE multipartIdentifier
         (ALTER | CHANGE) COLUMN? qualifiedName
         (TYPE dataType)? (COMMENT comment=STRING)? colPosition?        #alterTableColumn
-    | ALTER TABLE tableIdentifier partitionSpec?
+    | ALTER TABLE multipartIdentifier partitionSpec?
```

Comment:
hmm, the parser rule above seems to include this one?

Reply:
I did not combine two rules and two statements. A possible combine might have few Option fields and a logic to interpret it to v1/v2 cases, a bit mess it sounds like.

Reply:
The v2 rule
```
(ALTER | CHANGE) COLUMN? qualifiedName (TYPE dataType)? (COMMENT comment=STRING)?
```
The `colType`
```
colName=errorCapturingIdentifier dataType (COMMENT STRING)?
```

Seems like the v1 rule is more powerful and can rename a column and change data type together. If other DBs support it as well, maybe Spark should also support it. Otherwise, maybe 3.0 is a good time to drop this syntax.

Reply:
Seems we do not really support it, though we allow such syntax:

https://github.com/apache/spark/blob/7aca0dd658b8bda05574b3df3254aaf66eb2a174/sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala#L343-L351

Reply:
Not sure what if the original plan is to support that in the future.

Reply:
Let's simply remove this syntax. We have ALTER TABLE CHANGE COLUMN and ALTER TABLE RENAME COLUMN, which is good enough to me.

Reply:
ok. let's remove it.